p8e-api/src/main/kotlin/io/provenance/engine/event/IndexHandler.kt

```diff
@@ -114,6 +123,36 @@ class IndexHandler(
                                     envelope.data.result.contract.recitalsList.map { it.signer.encryptionPublicKey.toPublicKey().toSha512Hex() }
                                 ), RequestOptions.DEFAULT
                             )
+                            // If the env is the invoker, create the data access message and put into a job.
+                            if(envelope.isInvoker == true) {
```

Comment:
We could save some time too by adding `&& envelope.data.input.affiliateShareLists.isNotEmpty()`.